src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py

```diff
@@ -361,7 +365,18 @@ def save_rich_type(self, attribute, value):
     raise NotImplementedError('Child must implement.')
 
   def serialize(self):
-    return self.proto.SerializeToString()
+    """Serializes UworkerMsg into proto wire format."""
+    for attr in self._set_fields:
+      # Do this to make sure _proto fields are up to date.
```

Comment:
Can you explain how this can get out of sync? I'm wondering if this wrapper approach should be replaced by a dumber approach of just modelling the input/output classes in Python and serializing them to proto only when we need to send through through the wire. 

There's a lot of black magic happening here.

Reply:
I can leave a TODO. Titouan is exploring a new approach.